### ISSUE TITLE
$this->action is null in postProcess method

### ISSUE THREAD

**Issue by gfilippakis** [PrestaShop/PrestaShop#33084]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

$this->action is null in the first if block of the postProcess method in AdminController class.

This fact causes a PHP deprecation notice in PHP 8.1 stating that:

> ucfirst(): Passing null to parameter #1 ($string) of type string is deprecated

### Expected behavior

The expected behavior is calling all the hooks without any PHP notices or warnings.

### Steps to reproduce

1. Run your installation with PHP 8.1
2. Go to 'Modules and Services'
3. Check your error log

### PrestaShop version(s) where the bug happened

8.1.x

### PHP version(s) where the bug happened

PHP 8.1

### Your company or customer's name goes here (if applicable).

Sleed

**Comment by sallemiines** [comment]
Hello @gfilippakis 

You indicated that you are installing PS 1789 with PHP 8.1
this version of prestashop is not compatible with php 81 (  PrestaShop requires PHP 7.4 or lower)
Could you please check the given steps and provide us with a screenshot of the error you have!
Thank you !

**Comment by gfilippakis** [comment]
@sallemiines this bug actually affects ALL PrestaShop versions.

This is the error from the error log:

> [Tue Jul 04 00:20:28.259926 2023] [proxy_fcgi:error] [pid 375:tid 140162122098240] [client 192.168.100.56:59771] AH01071: Got error 'PHP message: PHP Deprecated:  ucfirst(): Passing null to parameter #1 ($string) of type string is deprecated in ...

The thing is that there is actually a wrong variable used inside the "if" block. The variable should be $action instead of $this->action. 

I submitted a [PR](https://github.com/PrestaShop/PrestaShop/pull/33083) yesterday regarding this, but then I had to associate an issue with the bug fix. So here we are.
